lib/Server.js

```diff
@@ -381,14 +381,22 @@ class Server {
                 }
               }
 
-              const bypass = typeof proxyConfig.bypass === 'function';
-
-              const bypassUrl =
-                (bypass && proxyConfig.bypass(req, res, proxyConfig)) || false;
-
-              if (bypassUrl) {
+              // - Check if we have a bypass function defined
+              // - In case the bypass function is defined we'll retrieve the
+              // bypassUrl from it otherwise byPassUrl would be null
+              const isByPassFuncDefined =
+                typeof proxyConfig.bypass === 'function';
+              const bypassUrl = isByPassFuncDefined
+                ? proxyConfig.bypass(req, res, proxyConfig)
+                : null;
+
+              if (typeof bypassUrl === 'boolean') {
```

Comment:
This is not aligning to the documentation 
> In the function you get access to the request, response and proxy options. It must return either false or a path that will be served instead of continuing to proxy the request.

Reply:
PR welcome to webpack docs

Reply:
@evilebottnawi https://github.com/webpack/webpack.js.org/pull/3211